mono/btls/CMakeLists.txt

```diff
@@ -80,12 +80,8 @@ set (
 	btls-x509-verify-param.h
 	btls-x509.c
 	btls-x509.h
-
-	${BORINGSSL_OBJECTS}
 )
 
-if (BUILD_SHARED_LIBS)
-	add_library (mono-btls-shared SHARED ${MONO_BTLS_SOURCES})
-else ()
-	add_library (mono-btls-static STATIC ${MONO_BTLS_SOURCES})
-endif ()
+add_library (mono-btls-objects OBJECT ${MONO_BTLS_SOURCES})
+add_library (mono-btls-shared SHARED $<TARGET_OBJECTS:mono-btls-objects> ${BORINGSSL_OBJECTS})
```

Comment:
The shared objects need to be built using flags like -fPIC. Is this going to pass those flags ?